### Align reads to assemblies in pairs

[To see previous alignment of reads to its own assemblies](https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/tree/master/Work_files/mahesh_assemblies_workfolder/Check_assemblies_map_reads).

[To see previous alignment of weird strain reads to other weird strain assemblies](https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/tree/master/Work_files/mahesh_assemblies_workfolder/Check_weird_strains).

This time, merging previously fastq files:

For weird strains done with: 

- https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/blob/master/Work_files/mahesh_assemblies_workfolder/Align_reads_assemblies/join_fastq_weird_strains.sh

For *Debaryomyces hansenii* strains done previously for the variant calling, found here:

- PCRduplicates removal: https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/blob/master/Work_files/Only_Debaryomyceshansenii/Variant_calling_4/1_pcrdupl.sh

- Merge files: https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/blob/master/Work_files/Only_Debaryomyceshansenii/Variant_calling_4/2_cat.sh

#### Pipeline for Bowtie2 in pairs:

Found [here](

