### ACT1

Sequences extracted from the assemblies using BLAST, found here:

https://github.com/The-Bioinformatics-Group/Debaryomyces_hansenii/blob/master/Work_files/Only_Debaryomyceshansenii/Phylogeny/genes/BLAST/README.md

Download sequences used for the article of Jacques *et al*., 2009:

http://ijs.microbiologyresearch.org/content/journal/ijsem/10.1099/ijs.0.004325-0

*Debaryomyces coudertii*

http://www.ncbi.nlm.nih.gov/nuccore/AJ867051

*Debaryomyces polymorphus* var. *polymorphus*

http://www.ncbi.nlm.nih.gov/nuccore/AJ604532

*Debaryomyces castelli*

http://www.ncbi.nlm.nih.gov/nuccore/AJ867052.1

*Debaryomyces vanrijiae* var. *vanrijiae*

http://www.ncbi.nlm.nih.gov/nuccore/AJ604534

*Debaryomyces pseudopolymorphus*

http://www.ncbi.nlm.nih.gov/nuccore/AJ604533

*Debaryomyces macquariensis*

http://www.ncbi.nlm.nih.gov/nuccore/FM165417

*Debaryomyces tyrocola*

http://www.ncbi.nlm.nih.gov/nuccore/AM992961

*Debaryomyces hansenii*

http://www.ncbi.nlm.nih.gov/nuccore/XM_458703

*Candida flareri*

http://www.ncbi.nlm.nih.gov/nuccore/AJ508464

*Debaryomyces fabryi*

http://www.ncbi.nlm.nih.gov/nuccore/AJ508504

*Debaryomyces maramus*

http://www.ncbi.nlm.nih.gov/nuccore/AJ606311

*Debaryomyces nepalensis*

http://www.ncbi.nlm.nih.gov/nuccore/FM165416

http://www.ncbi.nlm.nih.gov/nuccore/AJ867054


Join together with the act1 sequences of our strains:

```cat act1_deha.fasta act1_debaryomyces_sp.fasta >> act1_all.fasta```

Two types have to be done, exon2 only to compare with the rest of strains and complete sequence to compare D. hansenii strains.


#### More from glacier strains:

*CLIB 1348*

http://www.ncbi.nlm.nih.gov/nuccore/LK391880

*CLIB 1460*

http://www.ncbi.nlm.nih.gov/nuccore/LK391881

*CLIB 1462*

http://www.ncbi.nlm.nih.gov/nuccore/LK391882

*CLIB 1464*

http://www.ncbi.nlm.nih.gov/nuccore/LK391883

*CLIB 1466*

http://www.ncbi.nlm.nih.gov/nuccore/LK391884

*CLIB 1469*

http://www.ncbi.nlm.nih.gov/nuccore/LK391885

*CLIB1472*

http://www.ncbi.nlm.nih.gov/nuccore/LK391886

*CLIB 1479*

http://www.ncbi.nlm.nih.gov/nuccore/LK391887

*CLIB 1480*

http://www.ncbi.nlm.nih.gov/nuccore/LK391888

*CLIB 1486*

http://www.ncbi.nlm.nih.gov/nuccore/LK391889

*CLIB 1490*

http://www.ncbi.nlm.nih.gov/nuccore/LK391890

*CLIB 1277*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391891

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391892

*CLIB 1299*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391893

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391894

*CLIB 1300*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391895

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391896

*CLIB 1456*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391897

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391898

*CLIB 1458*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391899

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391900

*CLIB 1459*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391901

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391902

*CLIB 1461*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391903

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391904

*CLIB 1463*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391905

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391906

*CLIB 1467*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391907

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391908

*CLIB 1470*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391909

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391910

*CLIB 1471*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391911

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391912

*CLIB 1473*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391913

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391914

*CLIB 1474*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391915

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391916

*CLIB 1476*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391917

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391918

*CLIB 1477*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391919

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391920

*CLIB 1478*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391921

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391922

*CLIB 1481*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391923

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391924

*CLIB 1482*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391925

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391926

*CLIB 1483*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391927

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391928

*CLIB 1484*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391929

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391930

*CLIB 1485*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391931

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391932

*CLIB 1487*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391933

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391934

*CLIB 1491*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391935

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391936

*CLIB 1488*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391937

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391938

*CLIB 1489*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LK391939

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LK391940

*CBS 5167*

http://www.ncbi.nlm.nih.gov/nuccore/AM992975

*CBS 1795*

http://www.ncbi.nlm.nih.gov/nuccore/AM992965

*CLIB 660*

http://www.ncbi.nlm.nih.gov/nuccore/AM992974

*CLIB 1301*

http://www.ncbi.nlm.nih.gov/nuccore/FR799736

*CLIB 1302*

http://www.ncbi.nlm.nih.gov/nuccore/FR799738

*CLIB 949*

http://www.ncbi.nlm.nih.gov/nuccore/LM994820

*CBS 790*

http://www.ncbi.nlm.nih.gov/nuccore/EU816214

*CLIB 1492*

Allele A: http://www.ncbi.nlm.nih.gov/nuccore/LM994818

Allele B: http://www.ncbi.nlm.nih.gov/nuccore/LM994819

*CBS 792*

http://www.ncbi.nlm.nih.gov/nuccore/EU816211

*CBS 11666*

http://www.ncbi.nlm.nih.gov/nuccore/FR750248



### Phylogeny

Change identifiers of fasta file to avoid problems with the different programs:

```
sed 's/,/_/g' ACT1_Exon2_for_phylogeny.fasta > ACT1_Exon2_for_phylogeny_nocomas.fasta
sed 's/ /_/g' ACT1_Exon2_for_phylogeny_nocomas.fasta > ACT1_Exon2_for_phylogeny_nocomasnospaces.fasta
sed 's/:/_/g' ACT1_Exon2_for_phylogeny_nocomasnospaces.fasta > ACT1_Exon2_for_phylogeny_nocomasnospacesnocol.fasta
sed 's/-/_/g' ACT1_Exon2_for_phylogeny_nocomasnospacesnocol.fasta > ACT1_Exon2_for_phylogeny_nocomasnospacesnocolnolines.fasta
```

Also, the beginning of some names were changed because PhyML sometimes only shows 10 characters.

##### Muscle

```
muscle -in ACT1_Exon2_for_phylogeny_nocomasnospacesnocolnolines.fasta -out Muscle_ACT1_Exon2_for_phylogeny.fasta
```

##### Transform to .phy with ALTER

http://sing.ei.uvigo.es/ALTER/

##### ModelTest

jmodeltest-2.1.10

```
java -jar jmodeltest-2.1.10/jModelTest.jar -d Muscle_ACT1_Exon2_for_phylogeny.fasta.alter2.phy -g 4 -i -f -AIC -BIC -a > exon2_seq.log 2>&1
```

##### Run PhyML

- Bootstrap analysis

##### Look results on FigTree
